### PR TITLE
fix: Treeview menu loading perf issue- EXO-73169 - Meeds-io/MIPs#129

### PR DESCRIPTION
**Pull request by hakermi**
Treeview menu loading perf issue

### CODE REVIEW

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 1]
```diff
@@ -277,6 +277,22 @@ Page getNoteOfNoteBookByName(String noteType,
...
+  /**
+   * Check whether the page has children or not
+   *
+   * @param pageId target note id
+   * @return true if it has page and false if not
+   */
+  boolean hasChildren(long pageId);
+
+  /**
+   * Check if page has drafts
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java]:
```suggestion
   * Check if the given page has drafts or not
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 2]
```diff
@@ -277,6 +277,22 @@ Page getNoteOfNoteBookByName(String noteType,
...
+   *
+   * @param pageId target note id
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java]:
```suggestion
   * @param pageId note page id
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 3]
```diff
@@ -277,6 +277,22 @@ Page getNoteOfNoteBookByName(String noteType,
...
+   * @return true if page has drafts and false if not
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java]:
```suggestion
   * @return true if the given page has drafts and false if not
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 4]
```diff
@@ -277,6 +277,22 @@ Page getNoteOfNoteBookByName(String noteType,
    */
   List<Page> getChildrenNoteOf(Page note, boolean withDrafts, boolean withChild) throws WikiException;
 
+  /**
+   * Check whether the page has children or not
+   *
+   * @param pageId target note id
+   * @return true if it has page and false if not
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java]:
```suggestion
   * @return true if the given page has children and false if not
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 5]
```diff
@@ -277,6 +277,22 @@ Page getNoteOfNoteBookByName(String noteType,
    */
   List<Page> getChildrenNoteOf(Page note, boolean withDrafts, boolean withChild) throws WikiException;
 
+  /**
+   * Check whether the page has children or not
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java]:
```suggestion
   * Check if the given page has children or not
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java [hunk 6]
```diff
@@ -277,6 +277,22 @@ Page getNoteOfNoteBookByName(String noteType,
...
+   *
+   * @param pageId target note id
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/NoteService.java]:
```suggestion
   * @param pageId note page id
```

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java [hunk 7]
```diff
@@ -766,7 +766,23 @@ public List<Page> getChildrenNoteOf(Page note, boolean withDrafts, boolean withC
     }
     return pages;
   }
-  
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean hasChildren(long pageId) {
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java:774]: change dataStorage.hasChildren(pageId) in getChildrenNoteOf method with this method

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java [hunk 8]
```diff
@@ -1271,12 +1271,13 @@ public Response getFullTreeData(@Parameter(description = "Note path", required =
...
       responseData = getJsonTree(noteParam, context);
-      JsonNodeData rootNodeData = responseData.get(0);
+      JsonNodeData rootNodeData = responseData.getFirst();
       rootNodeData.setHasDraftDescendant(true);
       finalTree.add(rootNodeData);
       context.put(TreeNode.DEPTH, "1");
 
-      List<JsonNodeData> children = new ArrayList<>(rootNodeData.getChildren());
+      List<JsonNodeData> listChildren = rootNodeData.getChildren();
+      List<JsonNodeData> children = listChildren != null ? new ArrayList<>(listChildren) : new ArrayList<>();
```
Comment (azayati) [inline notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:1296]:
```suggestion
      List<JsonNodeData> children = rootNodeData.getChildren() != null ? rootNodeData.getChildren() : new ArrayList<>();
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java (hunk 8 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/rest/NotesRestService.java:1296]: The `new ArrayList<>(listChildren)` is to make the list mutable

File: notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java (hunk 7 above)
Comment (hakermi) [reply notes-service/src/main/java/org/exoplatform/wiki/service/impl/NoteServiceImpl.java:774]: Done

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**